app/controllers/katello/api/v2/content_view_versions_controller.rb

```diff
@@ -58,11 +58,15 @@ def show
     param :force_yum_metadata_regeneration, :bool, :desc => N_("Force metadata regeneration on the repositories " \
                                                            "in the content view version")
     def promote
-      is_force = ::Foreman::Cast.to_bool(params[:force])
-      task = async_task(::Actions::Katello::ContentView::Promote,
-                        @version, @environments, is_force, params[:description],
-                        :force_yum_metadata_regeneration => params[:force_yum_metadata_regeneration])
-      respond_for_async :resource => task
+      if @environments.all? { |e| e.organization_id == @version.organization.id }
+        is_force = ::Foreman::Cast.to_bool(params[:force])
+        task = async_task(::Actions::Katello::ContentView::Promote,
+                          @version, @environments, is_force, params[:description],
+                          :force_yum_metadata_regeneration => params[:force_yum_metadata_regeneration])
+        respond_for_async :resource => task
+      else
+        fail HttpErrors::BadRequest, _("Cannot promote to an environment out of organization")
+      end
```

Comment:
I think a better place for this might be in the ContentViewEnvironment model.  We could add a validator onto there so that when it saves, it simply checks to confirm that these two entites are in the same org and errors if not.  Then it would work even through other actions like IncrementalUpdate which can do a promote as part of it as well.